File: Android攻城狮的第二门课（第1季）----高级控件/app/src/main/java/com/example/xiaobao/android_one/home8/day3/Person.java
package com.example.xiaobao.android_one.home8.day3;

import java.io.Serializable;

/**
 * Created by xiaobao on 2020/7/1.
 */

public class Person implements Serializable {
    private String address;
    private int age;
    private String name;

    public Person(String address, int age, String name) {
        this.address = address;
        this.age = age;
        this.name = name;
    }

    @Override
    public String toString() {
        return "Person{" +
                "address='" + address + '\'' +
                ", age=" + age +
                ", name='" + name + '\'' +
                '}';
    }
}

File: xiaobaoKotlin/app/src/main/java/com/example/xiaobao/xiaobaokotlin/home13/day8/Person.kt
package com.example.xiaobao.xiaobaokotlin.home13.day8

/**
 * Created by xiaobao on 2020/7/15.
 */


//子类有主构造函数
open class Person(var name: String, var age: Int) {
    // 基类
   open fun study(){
        println("我毕业了")
    }

}

class Student(name: String, age: Int, var no: String, var score: Int) : Person(name, age) {

}


class Student2 : Person {
    /**次级构造函数**/
    constructor(name: String, age: Int, no: String, score: Int) : super(name, age) {

    }

    override fun study() {
        super.study()
        println("重写了 study")
    }

}
